src/System.Linq/src/System/Linq/Intersect.cs

```diff
@@ -19,8 +19,14 @@ public static IEnumerable<TSource> Intersect<TSource>(this IEnumerable<TSource>
             {
                 throw Error.ArgumentNull(nameof(second));
             }
-
-            return IntersectIterator(first, second, null);
+           
+            foreach (TSource element in first)
+            {
+                if (second.Contains(element))
```

Comment:
This calls `Contains` on `second` multiple times, without preparing a buffer first. There's no guarantee that the second call won't throw an exception, search an empty sequence, wait on a long-running database query before returning, search completely different values each time, access a webservice `first.Count()` times where before it was accessing it once, and so on.